olp-cpp-sdk-dataservice-read/src/CatalogClientImpl.cpp

```diff
@@ -40,12 +38,11 @@ namespace {
 constexpr auto kLogTag = "CatalogClientImpl";
 }
 
-CatalogClientImpl::CatalogClientImpl(
-    HRN hrn, std::shared_ptr<OlpClientSettings> settings,
-    std::shared_ptr<cache::KeyValueCache> cache)
-    : catalog_(std::move(hrn)),
-      settings_(settings),
-      api_client_(OlpClientFactory::Create(*settings)) {
+CatalogClientImpl::CatalogClientImpl(HRN catalog, OlpClientSettings settings)
+    : catalog_(std::move(catalog)),
+      settings_(std::make_shared<OlpClientSettings>(std::move(settings))) {
```

Comment:
Do we still need to keep it as shared_ptr?

Reply:
Yes, ApiRepository expects weak_ptr to settings and PrefetchTilesRepository with PrefetchTilesProvider expect shared_ptr to settings.

Reply:
We have a ticket to get rid of shared_ptr from these classes. PM me, if you want more details.